File: testdata/integration-scenarios/postconditionService.ts
/**
 * @ServiceSpec
 * operationId: "expectSuccessGetUser"
 * description: "期望成功的用户获取"
 * preconditions:
 *   "valid_method": {"==": [{"var": "http_method"}, "GET"]}
 * postconditions:
 *   "success_status": {"==": [{"var": "span.status.code"}, "OK"]}
 *   "ok_status": {"==": [{"var": "http_status_code"}, 200]}
 *   "has_user_data": {"!=": [{"var": "response_user"}, null]}
 */
export class PostconditionService {
    async expectSuccessGetUser(userId: string): Promise<User> {
        return this.userRepository.findById(userId);
    }
}

File: testdata/integration-scenarios/PostconditionService.java
/**
 * @ServiceSpec
 * operationId: "expectSuccessCreateUser"
 * description: "期望成功的用户创建"
 * preconditions:
 *   "valid_method": {"==": [{"var": "http_method"}, "POST"]}
 * postconditions:
 *   "success_status": {"==": [{"var": "span.status.code"}, "OK"]}
 *   "created_status": {"==": [{"var": "http_status_code"}, 201]}
 *   "has_user_id": {"!=": [{"var": "response_user_id"}, null]}
 */
public class PostconditionService {
    public User expectSuccessCreateUser(CreateUserRequest request) {
        return userRepository.save(new User(request));
    }
}
